core/src/impl/Kokkos_SimpleTaskScheduler.hpp

```diff
@@ -127,14 +132,18 @@ class ExecutionSpaceInstanceStorage
 
 template <class MemorySpace>
 class MemorySpaceInstanceStorage
-  : private NoUniqueAddressMemberEmulation<MemorySpace>
+  : private NoUniqueAddressMemberEmulation<MemorySpace, DefaultCtorNotOnDevice>
 {
 private:
 
-  using base_t = NoUniqueAddressMemberEmulation<MemorySpace>;
+  using base_t = NoUniqueAddressMemberEmulation<MemorySpace, DefaultCtorNotOnDevice>;
 
 protected:
 
+  MemorySpaceInstanceStorage()
+    : base_t()
+  { }
```

Comment:
Yes, except I wanted to be explicit about the fact that it's not available on the device.  If the user gets an error, they should be able to see that the `MemorySpaceInstanceStorage` default constructor isn't available on the device, and that will hopefully be (a little) more readable.  The `EBO` stuff is such a mess (albeit necessary) that I'd like to prevent error messages from trailing into that space.

Reply:
I meant
```C++
MemorySpaceInstanceStorage() = default;
```

Reply:
Yeah, I know :-)

You're right, I probably should have just done that.  But notice that this is the only constructor that's not device marked (i.e., `KOKKOS_INLINE_FUNCTION`), so it felt more natural to spell it out.  I think (not positive, though) that defaulted constructors might be device marked automatically if other constructors are?  Anyway, because of all of those things, my instinct was to just write it out.

Reply:
Yes defaulted constructors may end up being automatically device marked